### PR TITLE
[SURE-5388]fix: add owner reference to the secret

### PR DESCRIPTION
**Pull request by pratikjagrut**
## Problem
Custom downstream cluster secrets remain in the local cluster after the custom downstream cluster is deleted 
 
## Solution
Add owner reference to the secret so that the secret will be automatically deleted once the cluster is deleted.
 
### Manual Testing

- Create a custom rke1 cluster
- Check secret (kubectl -n cattle-system get secrets -l cattle.io/creator=norman) if it's created or not
- Delete the custom cluster
- Check secret (kubectl -n cattle-system get secrets -l cattle.io/creator=norman) if it's deleted or not

### Automated Testing

* Test types added/modified:
    * Unit
    * Integration (Go Framework)
    * Integration (v2prov Framework)
    * Validation (Go Framework)
    * Other - Explain: _EXPLAIN_
    * None
    * _REMOVE NOT APPLICABLE BULLET POINTS ABOVE_
* If "None" - Reason: _EXPLAIN THE REASON_
  
* If "None" - GH Issue/PR: _LINK TO GH ISSUE/PR TO ADD TESTS_

Summary: _TODO_

### Regressions Considerations

_TODO_

Existing / newly added automated tests that provide evidence there are no regressions:
* _TODO_

### CODE REVIEW

**Review by jiaqiluo — DISMISSED** [review]
LGTM.

It looks like the reason why this happens only in the RKE1 custom cluster is that Rancher skips the driver's removing which removes the secret. [code](https://github.com/rancher/rancher/blob/0ebc03cba3dce7ceac17cd88ccb5f16e974aea88/pkg/controllers/management/clusterprovisioner/provisioner.go#L159-L162), so adding the owner reference to the v3 Cluster will ensure the secret will be deleted.

**Review by jakefhyde — DISMISSED** [review]
Nice work!

File: pkg/encryptedstore/secrets_store.go [hunk 1]
```diff
@@ -65,18 +65,21 @@ func (g *GenericEncryptedStore) getKey(name string) string {
...
+func (g *GenericEncryptedStore) set(name string, data map[string]string, owner *metav1.OwnerReference) error {
 	logrus.Debugf("[GenericEncryptedStore]: set secret called for %v", g.getKey(name))
 	sec, err := g.secretLister.Get(g.namespace, g.getKey(name))
 	if errors.IsNotFound(err) {
 		logrus.Debugf("[GenericEncryptedStore]: Creating secret for %v", g.getKey(name))
 		sec = &corev1.Secret{}
 		sec.Name = g.getKey(name)
 		sec.StringData = data
+		if owner != nil {
+			sec.SetOwnerReferences([]metav1.OwnerReference{*owner})
```
Comment (kinarashah) [inline pkg/encryptedstore/secrets_store.go:81]: @jiaqiluo @jakefhyde Do you see issues with us also doing this for update so logic also applies for existing custom clusters?

**Comment by pratikjagrut** [comment]
@jiaqiluo, @kinarashah and @jakefhyde Any idea when this PR will be merged?

**Comment by pratikjagrut** [comment]
Should I change the base branch for this PR?

**Comment by jiaqiluo** [comment]
hi @pratikjagrut, the linked issue is not assigned to any milestone, and we are close to releasing Rancher v2.8.4, so I don't think we can merge this PR at the moment.   Could you work with your team or PMs to get the issue added to a milestone?

File: pkg/controllers/management/etcdbackup/etcdbackup.go [hunk 2]
```diff
@@ -55,13 +55,14 @@ type Controller struct {
 }
 
 func Register(ctx context.Context, management *config.ManagementContext) {
+	clusterClient := management.Management.Clusters("")
 	c := &Controller{
 		ctx:                   ctx,
-		clusterClient:         management.Management.Clusters(""),
```
Comment (jakefhyde) [inline pkg/controllers/management/etcdbackup/etcdbackup.go:60]: I would leave this as `management.Management.Clusters("")` for consistency.

**Review by jakefhyde — CHANGES_REQUESTED** [review]

**Comment by pratikjagrut** [comment]
@jakefhyde Updated the PR as per your review.

File: pkg/encryptedstore/secrets_store.go (hunk 1 above)
Comment (jakefhyde) [reply pkg/encryptedstore/secrets_store.go:81]: No I don't see any issues there. @pratikjagrut WDYT?

**Pull request closed without merging**